src/components/form/radio/radio.tsx

```diff
@@ -1,8 +1,29 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, {
+  FunctionComponent,
+  ChangeEventHandler,
+  HTMLAttributes,
+  ReactNode,
+} from 'react';
 import classNames from 'classnames';
+import { CommonProps } from '../../common';
 
-export const EuiRadio = ({
+export interface EuiRadioProps {
+  autoFocus?: boolean;
+  /**
+   * when `true` creates a shorter height radio row
+   */
+  compressed?: boolean;
+  label?: ReactNode;
+  name?: string;
+  value?: string;
+  checked?: boolean;
```

Comment:
The previous proptypes had `checked` as required; while the code works with it as a boolean, the component as a whole doesn't make sense to use without this prop, and we should keep it required.

```suggestion
  checked: boolean;
```

Reply:
Thanks for the feedback, I made a bunch of updates (with an assist from g2g)!

However, I am not certain on how to handle this one. I do see that it was previously required, but `checked` is not required in the sense that it is often omitted in the `false` (unchecked) state for radio inputs. This has me thinking that it perhaps ought to be optional?

Reply:
You're correct in that the code doesn't require it, but from a consumer's perspective it doesn't make sense to ever leave out: if disabled, you probably have an explicit state to represent; if enabled, you'll want the control toggleable so a value will be passed in regardless.

Making it required helps avoid accidental bugs when using typescript as it can't be left off, and if in vanilla javascript the proptypes get an important distinction and won't allow an `undefined` value to be [accidentally] passed.

Reply:
Ok, so when I switch that to required, my tests fail and I believe this is because `checked` is not explicitly passed to radio inputs that are in an unchecked state. I've done some digging, but do not know the path forward from here... how do I effectively say `checked` is required but then handled when it is not passed?

`<input type="radio" checked>` <-- checked
`<input type="radio">` <-- unchecked

Reply:
`checked={false}` would resolve the test issues. However, I dig a little more digging because the tests shouldn't have been working previously if not providing the required property. React has a list of special properties it handles in a different way, and `checked` is one of them. This means our `bool.isRequired` has never really applied, and so making this a required prop is potentially a breaking change, and is likely to be a headache.

With all of that, let's leave this as not required.